src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -1180,8 +1182,8 @@ void ShenandoahFreeSet::rebuild(size_t young_cset_regions, size_t old_cset_regio
 
   // Consult old-region surplus and deficit to make adjustments to current generation capacities and availability.
   // The generation region transfers take place after we rebuild.
-  size_t old_region_surplus = _heap->get_old_region_surplus();
-  size_t old_region_deficit = _heap->get_old_region_deficit();
+  size_t old_region_surplus = old_generation->get_region_surplus();
```

Comment:
To my earlier comment regarding carrying two variables one for +ve and one for -ve case, it looks like this would be text book case where a single variable with sign would shrink the code to half, make it more readable and get rid of the unnecessary if-then-else.

Not sure if it's also used somewhere else or how the values are set. May be worth a separate cleanup ticket.